scripts/lib/CIME/case/case_run.py

```diff
@@ -106,11 +110,13 @@ def _run_model_impl(case, lid, skip_pnl=False, da_cycle=0):
         stat = run_and_log_case_status(run_func, "model execution", caseroot=case.get_value("CASEROOT"))
         model_logfile = os.path.join(rundir, model + ".log." + lid)
         # Determine if failure was due to a failed node, if so, try to restart
-        if node_fail_re:
```

Comment:
We can have two different "counters" here, one for node fails and one for mpirun retries.